third_party/jitterentropy/jitterentropy-base-user.h

```diff
@@ -112,14 +112,24 @@ static inline void jent_get_nstime(uint64_t *out)
 	*out = EAX_EDX_VAL(val, low, high);
 }
 
-#else /* (__x86_64__) || (__i386__) */
+#elif defined(__aarch64__)
+
+static inline void jent_get_nstime(uint64_t *out)
+{
+        uint64_t ctr_val;
+        /*
+         * Use the system counter for aarch64 (64 bit ARM).
+         */
+        __asm__ volatile("mrs %0, cntvct_el0" : "=r" (ctr_val));
+        *out = ctr_val;
+}
+
+#else /* (__x86_64__) || (__i386__) || (__aarch64__) */
 
 static inline void jent_get_nstime(uint64_t *out)
 {
-	/* OSX does not have clock_gettime -- taken from
-	 * http://developer.apple.com/library/mac/qa/qa1398/_index.html */
 # ifdef __MACH__
-	*out = mach_absolute_time();
+	*out = clock_gettime_nsec_np(CLOCK_MONOTONIC_RAW);
```

Comment:
Is there a URL explaining `mach_absolute_time()` vs `clock_gettime_nsec_np()` we could link to? Or just `clock_gettime_nsec_np()` docs would probably be fine too.

Reply:
Adding this to a comment there is enough IMHO: https://developer.apple.com/documentation/kernel/1462446-mach_absolute_time

Reply:
done